### PR TITLE
PXD-1670 feat(clinical-download): enable download clinical data

### PR DESCRIPTION
**Pull request by qingyashu**

### CODE REVIEW

File: src/DataExplorer/utils.js [hunk 1]
```diff
@@ -57,14 +81,100 @@ const queryFileMappingData = async (apiFunc, projectId, selectedTableRows, arran
...
+    endpoint: `/${projectId}/graphql`,
+    body,
+  });
+  const columns = response.data[arrangerConfig.graphqlField].columnsState.state.columns
+    .filter(col => col.show)
+    .map(col => col.field);
+  return columns;
+};
+
+const querySelectedData = async (
```
Comment (abgeorge7) [inline src/DataExplorer/utils.js]: could this function, `downloadData`, and `queryFileMapping` just be a little more descriptive? This file is pretty dense so I think it'd make it more readable.

File: src/DataExplorer/utils.test.js [hunk 2]
```diff
@@ -48,7 +61,7 @@ describe('utils for downloading manifest', () => {
                     edges {
                           node {
                             uuid
-                            subject_id
+subject_id
```
Comment (abgeorge7) [inline src/DataExplorer/utils.test.js]: picky but re-space this 😬

File: src/DataExplorer/index.jsx [hunk 3]
```diff
@@ -17,7 +17,10 @@ class DataExplorer extends React.Component {
           projectId={arrangerConfig.projectId}
         >
           <DataExplorerFilters arrangerConfig={arrangerConfig} />
-          <DataExplorerVisualizations arrangerConfig={arrangerConfig} />
+          <DataExplorerVisualizations
+            arrangerConfig={arrangerConfig}
+            explorerTableConfig={arrangerConfig.table}
```
Comment (abgeorge7) [inline src/DataExplorer/index.jsx]: will this break if theres no `arrangerConfig` ?

File: src/DataExplorer/utils.js [hunk 4]
```diff
@@ -35,19 +53,46 @@ export const constructGraphQLQuery = (selectedTableRows, arrangerConfig, isGetti
...
+*       manifestMapping.fileIndexType - type name of file index
+*       manifestMapping.fileReferenceIdField - field name of reference field in file index
+*/
+const queryFileMappingData = async (apiFunc, projectId, idList, arrangerConfig) => {
   const countQuery = await apiFunc({
     endpoint: `/${projectId}/graphql`,
-    body: constructGraphQLQuery(selectedTableRows, arrangerConfig, true),
+    body: constructGraphQLQuery(
+      idList,
+      arrangerConfig.manifestMapping.fileIndexType,
```
Comment (abgeorge7) [inline src/DataExplorer/utils.js]: same here, what if there is no `manifestMapping` ?

**Comment by qingyashu** [comment]
I have added util function `hasKeyChain` and some checking for empty/error arrangerConfig, please have a review again~ Thanks @abgeorge7

**Pull request merged**